hack/_azure-set-vars.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+set -o errexit -o pipefail
+
+: "${AZURE_DEVOPS_EXT_PAT:?}"
+: "${AZURE_VARIABLE_GROUP_NAME:?}"
+: "${AZURE_ORGANIZATION:?}"
+: "${AZURE_PROJECT:?}"
+
+# Set xtrace after checking variables to avoid logging the PAT
+set -o xtrace
+
+# Needed for Azure Pipelines related functionality
+az extension add --name azure-devops
+
+az devops configure --defaults \
+    organization="https://dev.azure.com/$AZURE_ORGANIZATION" \
+    project="$AZURE_PROJECT"
+
+get_or_create_vargroup() {
+    local group_name=$1
+
+    local group_id
+    group_id=$(
+        az pipelines variable-group list --group-name "$group_name" --query '[0].id'
+    )
+
+    if [[ -z $group_id ]]; then
+        group_id=$(
+            az pipelines variable-group create \
+                --name "$group_name" \
+                --variables unused="need at least one variable in group" \
+                --authorize true \
+                --query id
+        )
+    fi
+
+    echo "$group_id"
+}
+
+VARGROUP_ID=$(get_or_create_vargroup "$AZURE_VARIABLE_GROUP_NAME")
+if [[ -z $VARGROUP_ID ]]; then
+    echo "Variable group creation sometimes fails despite succeeding, trying again..." >&2
+    VARGROUP_ID=$(get_or_create_vargroup "$AZURE_VARIABLE_GROUP_NAME")
```

Comment:
If it's still empty here, exit with an error?

Reply:
Fixing in #131